sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -79,13 +80,20 @@ class CacheManager extends Logging {
       logWarning("Asked to cache already cached data.")
     } else {
       val sparkSession = query.sparkSession
-      val qe = sparkSession.sessionState.executePlan(planToCache)
-      val inMemoryRelation = InMemoryRelation(
-        sparkSession.sessionState.conf.useCompression,
-        sparkSession.sessionState.conf.columnBatchSize, storageLevel,
-        qe.executedPlan,
-        tableName,
-        optimizedPlan = qe.optimizedPlan)
+      val originalValue = sparkSession.sessionState.conf.getConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED)
+      val inMemoryRelation = try {
```

Comment:
nit: `val qe = try ...` can reduce the code diff

Reply:
Here we need put `inMemoryRelation `in `try `not  `qe`. Because only when executing `qe.executedPlan`, it will be inserted `AdaptiveSparkPlanExec`. And I will remove the `qe `before `try` later.